### PR TITLE
Standardize the health probe and metrics arguments of scheduler.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
**What type of PR is this?**

/kind cleanup
Part of https://github.com/karmada-io/karmada/issues/5219

/assign @whitewindmills @RainbowMango

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-scheduler`:  Add health probe argument `health-probe-bind-address` and metrics argument `metrics-bind-address`. Deprecate `--bind-address` and `--secure-port` flags.
```

### CODE REVIEW

File: charts/karmada/templates/karmada-scheduler.yaml [hunk 1]
```diff
@@ -50,8 +50,8 @@ spec:
           command:
             - /bin/karmada-scheduler
             - --kubeconfig=/etc/kubeconfig
-            - --bind-address=0.0.0.0
-            - --secure-port=10351
+            - --metrics-bind-address=0.0.0.0:10351
+            - --health-probe-bind-address=0.0.0.0:10351
```
Comment (RainbowMango) [inline charts/karmada/templates/karmada-scheduler.yaml:54]: Probably the same reason as https://github.com/karmada-io/karmada/pull/5290#issuecomment-2311992249. 
We have to revert charts changes to get CI pass.

Comment (liangyuanpeng) [reply charts/karmada/templates/karmada-scheduler.yaml]: updated,Thanks!

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**